core/src/outputs.rs

```diff
@@ -138,3 +138,15 @@ fn are_valid_elements(outputs: &[u64]) -> bool {
     }
     true
 }
+
+impl Serializable for StackOutputs {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // stack
+        target.write_u64(self.stack.len() as u64);
+        self.stack.iter().copied().for_each(|v| target.write_u64(v));
+
+        // overflow addrs
+        target.write_u64(self.overflow_addrs.len() as u64);
+        self.overflow_addrs.iter().copied().for_each(|v| target.write_u64(v));
```

Comment:
We should be able to infer the number of overflow addresses from the number of output values. I believe it is something like: number of values - 16 (but I'm not 100% sure).

We don't need to do this in this PR - but I'd leave a TODO here for the future.

Also, the number will never be greater than `u32` and probably will never be greater than `u16` either. We could go with `u16` here for length - but then we'd need to panic if for w/e reason it is greater. So, maybe for now lets leave a TODO to address this in the future.

Reply:
This intersects with the serialization refactor that we probably need to work on the near future. The serialization from winterfell, at least by default, won't serialize the length of the arrays - so this is divergent by default.

The main problem here is that this function is infallible, so the only option is to panic. We can leave as `u32` for now with a TODO to refactor when we take the serialization normalization.